testcases/kernel/syscalls/utimensat/utimensat01.c

```diff
@@ -278,5 +335,245 @@ int main(int argc, char *argv[])
+/* Command line: ./utimensat01 -d ./dirfd_file ./utimesat01_testfile */
+char *argv_ptr6[] = {
+    (char *) "utimensat01",	/* Executable */
+    (char *) "-d",
+    (char *) "./dirfd_file",
+    (char *) "./utimesat01_testfile",
+    NULL
+};
+
+/* Test 8 - EINVAL: Invalid value in one or both of the tv_nsec fields. */
+/* Command line: ./utimensat01 ./utimesat01_testfile 1 -1 1 -1 */
+char *argv_ptr7[] = {
+    (char *) "utimensat01",	/* Executable */
+    (char *) "./utimesat01_testfile",
+    (char *) "1",
+    (char *) "-1",
+    (char *) "1",
+    (char *) "-1",
+    NULL
+};
+
+/* Test 9 - ENAMETOOLONG: pathname is too long (Long file name as argument */
+/* Command line: ./utimensat01 ./abcdef...... */
+char *argv_ptr8[] = {
+    (char *) "utimensat01",	/* Executable */
+    (char *) "./abcdefghijklmnopqrstuvwxaaaaaaaaaaaaaaaaaaaaaaaaaa \
+                aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa \
+                aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa \
+                aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+    NULL
+};
+
+/* Test 10 - ENOENT (utimensat()) A component of pathname does not refer
+*  to an existing directory or file, or pathname is an empty string.
+*/
+/* Command line: ./utimensat01  (An empty string as file name) */
+char *argv_ptr9[] = {
+    (char *) "utimensat01",	/* Executable */
+    (char *) "",
+    NULL
+};
+
+/* Test arguments for each of test case 1 to 10 */
+char **argv_ptr[] = {argv_ptr0, argv_ptr1, argv_ptr2, argv_ptr3, argv_ptr4, 
+	             argv_ptr5, argv_ptr6, argv_ptr7, argv_ptr8, argv_ptr9};
+
+/* Test arguments size for each of test case 1 to 10 */
+int  argv_size[]  = {2, 6, 6, 6, 6, 6, 4, 6, 2, 2};
+
+
+/* Expected test result for each of test case 1 to 10.
+ * 0 - Expecting result is success i.e. 0 from new_main() function.
+ * Other error numbers - Expected corresponding errorno.
+ */
+int  expected_result[] = {0, 0, 0, 0, 0, 0, EBADF, EINVAL, ENAMETOOLONG, ENOENT}; 
+
+/* Newly added main() function to pass on relevant argc and argv to
+ * new_main() based on the test case and check the pass or failure
+ * of utimensat(). This is just a stub() function to use the above
+ * argc and argv arrays and invoke new_main() for performing the test.
+ *
+ * It does the following.
+ * 	- Invoke setup() function to create a test file.
+ * 	- For each test case,
+ * 		1. reset the time stamps of test file using reset_tsp() function.
+ * 		2. invoke new_main() with proper argc and argv as per test case.
+ * 		3. validate the return value of new_main() to check if the test is
+ * 		   passed or failed.
+ */
+
+int main(int argc, char *argv[])
+{
+	int i;
+	int ret = -1;
+
+	ret = setup(O_CREAT | O_RDONLY, S_IRUSR, "./utimesat01_testfile");
+	if (ret < 0)
+	{
+		tst_brkm(TCONF, NULL, "Failed to create test file");
+	}
+
+	/* Tests 1 to 10 */
+	for (i = 0; i < 10; i++)
+	{
+		ret = reset_tsp ();
```

Comment:
Instead of create, delete the file for each test, the reset will simply change the file time stamp to Jan 1, 1970 (system epoch time). The prints at each end of test if enabled will give a clue on the reset of time stamp and the effect of changing it using the utimensat api..